### ISSUE TITLE
pointerEvents="box-none" does not consistently allow VoiceOver activation of sibling Touchable backdrop element

### ISSUE THREAD

**Issue by dcalhoun** [facebook/react-native#32759]
### Description

When attempting to build a "backdrop" element which sits behind content and allows dismissing the content on tap — e.g. a modal dialog's backdrop — setting `pointerEvents="box-none"` on the content container does not consistently allow for VoiceOver to activate the backdrop's `onPress` callback. 

The height of the content container appears to impact whether one can successfully activate the backdrop's `onPress` callback with VoiceOver. If the content container is small — e.g. less than 30% of the screen height — the `onPress` callback successfully activates. If the content container is large — e.g. 85% of screen height — the `onPress` callback _does not_ successfully activate. 

### Version

0.64.3

### Output of `react-native info`

```
System:
    OS: macOS 11.6.1
    CPU: (16) x64 Intel(R) Core(TM) i9-9980HK CPU @ 2.40GHz
    Memory: 117.29 MB / 32.00 GB
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 17.2.0 - ~/.asdf/shims/node
    Yarn: 1.22.17 - /usr/local/bin/yarn
    npm: 8.1.4 - ~/.asdf/shims/npm
    Watchman: 2021.12.13.00 - /usr/local/bin/watchman
  Managers:
    CocoaPods: 1.11.2 - /usr/local/bin/pod
  SDKs:
    iOS SDK:
      Platforms: DriverKit 21.0.1, iOS 15.0, macOS 12.0, tvOS 15.0, watchOS 8.0
    Android SDK:
      API Levels: 28, 29, 30, 31
      Build Tools: 28.0.3, 29.0.2, 30.0.2
      System Images: android-29 | Intel x86 Atom, android-30 | Google APIs Intel x86 Atom
      Android NDK: Not Found
  IDEs:
    Android Studio: 2020.3 AI-203.7717.56.2031.7784292
    Xcode: 13.1/13A1030d - /usr/bin/xcodebuild
  Languages:
    Java: 1.8.0_292 - /usr/bin/javac
  npmPackages:
    @react-native-community/cli: Not Found
    react: 17.0.1 => 17.0.1
    react-native: 0.64.3 => 0.64.3
    react-native-macos: Not Found
  npmGlobalPackages:
    *react-native*: Not Found
```

### Steps to reproduce

1. Open the included reproduction Snack with the iOS Expo Go app. 
2. Enable VoiceOver on iOS. 
3. Open the "short" view/modal using VoiceOver. 
4. Move VoiceOver focus to the semi-transparent backdrop. 
5. Double-tap to activate the backdrop `onPress` and dismiss the view/modal. 
6. Note the view/modal closed. 
7. Open the "long" view/modal using VoiceOver. 
8. Move VoiceOver focus to the semi-transparent backdrop. 
9. Double-tap to activate the backdrop `onPress` and dismiss the view/modal. 
10. ⚠️ Note the view/modal did not close. 

### Snack, code example, screenshot, or link to a repository

I created a [Snack reproduction](https://snack.expo.dev/@davidcalhoun/pointerevents-voiceover-compatibility) based upon the React Native [Modal documentation](https://reactnative.dev/docs/modal). 

<details><summary>Inlined Example Code</summary>

```jsx
import React, { useState } from 'react';
import {
  Alert,
  Dimensions,
  Modal,
  Pressable,
  StyleSheet,
  Text,
  TouchableWithoutFeedback,
  View,
} from 'react-native';

const ContentWithBackdrop = ({ onClose, height }) => {
  return (
    <>
      <TouchableWithoutFeedback
        accessibilityLabel="Dismiss dialog"
        accessibilityRole="button"
        onPress={onClose}>
        <View
          style={[
            styles.centeredView,
            styles.backdrop,
            StyleSheet.absoluteFill,
          ]}
        />
      </TouchableWithoutFeedback>
      <View pointerEvents="box-none" style={styles.centeredView}>
        <View style={[styles.modalView, { height }]}>
          <Text style={styles.modalText}>Hello World!</Text>
          <Pressable
            style={[styles.button, styles.buttonClose, { marginBottom: 16 }]}
            onPress={() => Alert.alert('Hello!')}>
            <Text style={styles.textStyle}>Say Hello</Text>
          </Pressable>
          <Pressable
            style={[styles.button, styles.buttonClose]}
            onPress={onClose}>
            <Text style={styles.textStyle}>Hide Modal</Text>
          </Pressable>
        </View>
      </View>
    </>
  );
};

const App = () => {
  const { height } = Dimensions.get('screen');

  const [shortModalVisible, setShortModalVisible] = useState(false);
  const [tallModalVisible, setTallModalVisible] = useState(false);
  const [shortViewVisible, setShortViewVisible] = useState(false);
  const [tallViewVisible, setTallViewVisible] = useState(false);

  return (
    <View style={styles.centeredView}>
      <Modal
        animationType="slide"
        transparent={true}
        visible={shortModalVisible}
        onRequestClose={() => {
          Alert.alert('Modal has been closed.');
          setShortModalVisible(false);
        }}>
        <ContentWithBackdrop onClose={() => setShortModalVisible(false)} />
      </Modal>
      <Modal
        animationType="slide"
        transparent={true}
        visible={tallModalVisible}
        onRequestClose={() => {
          Alert.alert('Modal has been closed.');
          setTallModalVisible(false);
        }}>
        <ContentWithBackdrop
          onClose={() => setTallModalVisible(false)}
          height={height * 0.85}
        />
      </Modal>

      <Pressable
        style={[styles.button, styles.buttonOpen, { marginBottom: 16 }]}
        onPress={() => setShortModalVisible(true)}>
        <Text style={styles.textStyle}>Show Short Modal</Text>
      </Pressable>
      <Pressable
        style={[styles.button, styles.buttonOpen, { marginBottom: 16 }]}
        onPress={() => setTallModalVisible(true)}>
        <Text style={styles.textStyle}>Show Tall Modal</Text>
      </Pressable>
      <Pressable
        style={[styles.button, styles.buttonOpen, { marginBottom: 16 }]}
        onPress={() => setShortViewVisible(true)}>
        <Text style={styles.textStyle}>Show Short View</Text>
      </Pressable>
      <Pressable
        style={[styles.button, styles.buttonOpen]}
        onPress={() => setTallViewVisible(true)}>
        <Text style={styles.textStyle}>Show Tall View</Text>
      </Pressable>

      {shortViewVisible && (
        <View style={StyleSheet.absoluteFill}>
          <ContentWithBackdrop onClose={() => setShortViewVisible(false)} />
        </View>
      )}
      {tallViewVisible && (
        <View style={StyleSheet.absoluteFill}>
          <ContentWithBackdrop
            height={height * 0.85}
            onClose={() => setTallViewVisible(false)}
          />
        </View>
      )}
    </View>
  );
};

const styles = StyleSheet.create({
  centeredView: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    marginTop: 22,
  },
  backdrop: {
    flex: 1,
    opacity: 0.5,
    backgroundColor: 'black',
  },
  modalView: {
    margin: 20,
    backgroundColor: 'white',
    borderRadius: 20,
    padding: 35,
    alignItems: 'center',
    shadowColor: '#000',
    shadowOffset: {
      width: 0,
      height: 2,
    },
    shadowOpacity: 0.25,
    shadowRadius: 4,
    elevation: 5,
  },
  button: {
    borderRadius: 20,
    padding: 10,
    elevation: 2,
  },
  buttonOpen: {
    backgroundColor: '#F194FF',
  },
  buttonClose: {
    backgroundColor: '#2196F3',
  },
  textStyle: {
    color: 'white',
    fontWeight: 'bold',
    textAlign: 'center',
  },
  modalText: {
    marginBottom: 15,
    textAlign: 'center',
  },
});

export default App;
```

</details>

https://user-images.githubusercontent.com/438664/146073507-53b104d7-93fc-4341-9ef6-9f2dfac7e4cc.MP4

**Comment by dcalhoun** [comment]
I believe this to still be a relevant issue.
